keras_cv/models/adain.py

```diff
@@ -0,0 +1,228 @@
+        """Computes the AdaIn feature map.
+
+        Args:
+            style: The style feature map.
+            content: The content feature map.
+
+        Returns:
+            The AdaIN feature map.
+        """
+        style, content = inputs
+        content_mean, content_std = get_mean_std(content)
+        style_mean, style_std = get_mean_std(style)
+        t = style_std * (content - content_mean) / content_std + style_mean
+        return t
+
+
+class AdaINModel(keras.Model):
+    def __init__(self, image_size=(None, None), style_weight=4.0, **kwargs):
+        super().__init__(**kwargs)
+        self.image_size = image_size
+        self.encoder = get_mini_vgg19_encoder(image_size)
+        self.decoder = get_adain_decoder()
+        self.adain_layer = AdaIN()
+        self.loss_net = get_loss_net(image_size)
+        self.style_weight = style_weight
+
+    def produce_functional_model():
+        style_input = layers.Input([*self.image_size, 3])
+        style_encoded = self.encoder(style_input)
+        content_input = layers.Input([*self.image_size, 3])
+        content_encoded = self.encoder(content_input)
+
+        adain_feature_map = self.adain_layer((style_encoded, content_encoded))
+        content_decoded = self.decoder(adain_feature_map)
+        
+        return keras.Model([style_input, content_input], content_decoded, name="adain_style_transfer")
+
+
+    def compile(self, optimizer, loss_fn):
+        super().compile()
+        self.optimizer = optimizer
+        self.loss_fn = loss_fn
+        self.style_loss_tracker = keras.metrics.Mean(name="style_loss")
+        self.content_loss_tracker = keras.metrics.Mean(name="content_loss")
+        self.total_loss_tracker = keras.metrics.Mean(name="total_loss")
+        
+    def call(self, inputs):
+        style, content = inputs
+        # Encode the style and content image.
+        style_encoded = self.encoder(style)
+        content_encoded = self.encoder(content)
+
+        # Compute the AdaIN target feature maps.
+        t = self.adain_layer((style_encoded, content_encoded))
+
+        # Generate the neural style transferred image.
+        return self.decoder(t), t
+        
+
+    def train_step(self, inputs):
+        style, content = inputs
+
+        # Initialize the content and style loss.
+        loss_content = 0.0
+        loss_style = 0.0
+
+        with tf.GradientTape() as tape:
+            reconstructed_image, t = self(inputs)
+            # Compute the losses.
+            reconstructed_vgg_features = self.loss_net(reconstructed_image)
+            style_vgg_features = self.loss_net(style)
+            loss_content = self.loss_fn(t, reconstructed_vgg_features[-1])
+            for inp, out in zip(style_vgg_features, reconstructed_vgg_features):
+                mean_inp, std_inp = get_mean_std(inp)
+                mean_out, std_out = get_mean_std(out)
+                loss_style += self.loss_fn(mean_inp, mean_out) + self.loss_fn(
```

Comment:
you'll be able to update this to `self.compiled_loss(mean_inp, mean_out) + self.compiled_loss(std_inp, std_out)`